File: PoZiomkaDomain/Student/Dtos/StudentModel.cs
namespace PoZiomkaDomain.Student.Dtos;


public record StudentModel(int Id, string Email, string? FirstName, string? LastName, string PasswordHash,
    bool IsConfirmed, string? PhoneNumber, string? IndexNumber,
    int? ReservationId, bool? HasAcceptedReservation, bool IsPhoneNumberHidden, bool IsIndexNumberHidden)
{

    public StudentDisplay ToStudentDisplay(bool hidePersonalInfo) => new(
        Id, Email, FirstName, LastName,
        hidePersonalInfo && IsIndexNumberHidden ? null : IndexNumber,
        hidePersonalInfo && IsPhoneNumberHidden ? null : PhoneNumber,
        ReservationId, HasAcceptedReservation
    );
};

File: PoZiomkaApp/src/app/student/student.model.ts
export type StudentModel = {
  id: number,
  email: string,
  firstName?: string,
  lastName?: string,
  indexNumber?: string,
  phoneNumber?: string,
  reservationId?: number,
  hasAcceptedReservation?: boolean,
  roomId?: number,
  canFillForms?: boolean
  isPhoneNumberHidden: boolean;
  isIndexNumberHidden: boolean;
}

export type StudentCreate = {
  email: string,
  firstName?: string,
  lastName?: string,
  indexNumber?: string,
  phoneNumber?: string
}

export type StudentUpdate = {
  id: number;
  firstName?: string;   
  lastName?: string;    
  indexNumber?: string; 
  phoneNumber?: string; 
  isPhoneNumberHidden: boolean;
  isIndexNumberHidden: boolean;
};
